sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsConf.scala

```diff
@@ -1632,13 +1632,34 @@ object RapidsConf {
     .bytesConf(ByteUnit.BYTE)
     .createWithDefault(500 * 1024)
 
+  val IO_COMPRESSION_CODEC = conf("spark.rapids.io.compression.codec")
+    .doc("The CPU compression codec used to compress data at the rest for both shuffle and " +
+      "spill. Supported codecs: lz4, lzf, snappy, zstd")
+    .startupOnly()
+    .stringConf
+    .createWithDefault("lz4")
+
+  val SHUFFLE_COMPRESSION_ENABLED = conf("spark.rapids.shuffle.io.compression.enabled")
```

Comment:
Why do we have separate configs instead of using the existing config Spark uses for shuffle and spill configuration?  The problem with separate configs is if the user has setup shuffle compression and/or encryption before, they will _not_ get it in practice when using the RAPIDS Accelerator which is surprising.

We should leverage the Spark configs.  I'd be OK with using custom configs as an override for the Spark configs iff the custom configs are set. However we should not add those configs unless we know we absolutely need them.

Reply:
@winningsix I would love to see some performance numbers for this too. I'm happy to help produce them if you want me to. For encryption I think we have to follow what Spark has set no matter what. I see no reason to have a separate config for that. It is a security problem if we don't follow the Spark config. But for compression I can see us having some leeway, and I would love some hard numbers to back up any decision we make about it.

Reply:
I think on the performance the thought is that if we are compressing on the host we are writing less to disk, so we may come up even or faster, depending on the compression ratio. So the idea was to run a case that would spill terabytes with and without compression, to see what the effects are.

In terms of configs, yes agree the Spark configs seem a bit confusing. I do like that broadcasts can be optionally not compressed, so that could be follow on work to figure out how to implement that. Here are some ideas:

- For the codec, we should just use the default from Spark, specified here: `spark.io.compression.codec`. You added `spark.rapids.shuffle.compression.codec` for shuffle, but I don't see a way to control the codec specifically for shuffle in Spark. In the name of less configs I propose we remove that too, and just use `spark.io.compresion.codec`.

- `spark.rapids.shuffle.io.compression.enabled` doesn't make a lot of sense right now. We have nvcomp right now for UCX shuffle compression and the buffers that you picked for compression are UCX buffers (`ShuffleBufferId` and `ShuffleReceivedBufferId` are exclusively for UCX).

- In order to enable/disable compression, because Spark has `spark.broadcast.compress`, `spark.shuffle.compress`, `spark.rdd.compress`, etc, I don't think there's a great mapping to our stuff. Instead of all of this, I think the only config we add is `spark.rapids.spill.io.compression.enabled` and that just says whether we will use `spark.io.compression.codec` to compress and decompress spill blocks that are NOT UCX shuffle.

We can revisit the UCX shuffle pieces at a later time, but I think that's outside of the scope.

Reply:
>  I think the only config we add is spark.rapids.spill.io.compression.enabled and that just says whether we will use spark.io.compression.codec to compress and decompress spill blocks that are NOT UCX shuffle.

I don't think we want to separate this.  Spill is spill.  If compression helps the CPU, it's very likely to help the GPU.  I don't think we should add this config until we have performance numbers justifying it.  Even if we do add it, we clearly have use cases that are failing today when it's not compressing, so arguably `false` is not the best default for it.

Reply:
Discussed with @jlowe  Respecting Spark existing configuration sounds good. Dropped the newly added configuration in latest update.